freeipa/src/main/java/com/sequenceiq/freeipa/flow/chain/ModifyProxyConfigFlowEventChainFactory.java

```diff
@@ -0,0 +1,32 @@
+package com.sequenceiq.freeipa.flow.chain;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.flow.core.chain.FlowEventChainFactory;
+import com.sequenceiq.flow.core.chain.config.FlowTriggerEventQueue;
+import com.sequenceiq.freeipa.flow.stack.modify.proxy.event.ModifyProxyFlowChainTriggerEvent;
+import com.sequenceiq.freeipa.flow.stack.update.UpdateUserDataEvents;
+import com.sequenceiq.freeipa.flow.stack.update.event.UserDataUpdateRequest;
+
+@Component
+public class ModifyProxyConfigFlowEventChainFactory implements FlowEventChainFactory<ModifyProxyFlowChainTriggerEvent> {
+
+    @Override
+    public String initEvent() {
+        return FlowChainTriggers.MODIFY_PROXY_CHAIN_TRIGGER_EVENT;
+    }
+
+    @Override
+    public FlowTriggerEventQueue createFlowTriggerEventQueue(ModifyProxyFlowChainTriggerEvent event) {
+        Queue<Selectable> flowEventChain = new ConcurrentLinkedQueue<>();
+        flowEventChain.add(new UserDataUpdateRequest(UpdateUserDataEvents.UPDATE_USERDATA_TRIGGER_EVENT.event(), event.getResourceId(), event.accepted())
```

Comment:
why is this a flowchain if it has 1 element?

Reply:
it starts another flow, and also another one will be needed for salt update when it will be extended to existing freeipa instances

Reply:
I don't get the first part, but if you are planning another flow in a followup PR, then it's ok. Commit message doesn't include this info